cmd/prometheus/uname_default.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// +build !linux
+
+package main
+
+import "runtime"
+
+// Uname for any platform other than linux.
+func Uname() string {
+	return "(Printing uname isn't yet supported on ", runtime.GOOS, ")"
```

Comment:
I'd just print the OS, rather than confusing the user about what is/isn't supported.

This feature is more for us as developers providing support than for end users.